### PR TITLE
Added query for entire dataset + tests

### PR DESCRIPTION
**Pull request by daiLlew**
Added check for empty filter & new query to get all the data

### CODE REVIEW

File: observation/store_test.go [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package observation_test
 
 import (
+	"fmt"
 	"github.com/ONSdigital/dp-filter/observation"
 	"github.com/ONSdigital/dp-filter/observation/observationtest"
 	"github.com/johnnadratowski/golang-neo4j-bolt-driver"
+	"github.com/satori/go.uuid"
```
Comment (Matt-Guest) [inline observation/store_test.go]: Do you mind updating the vendor dependencies?

File: observation/store.go [hunk 2]
```diff
@@ -31,15 +32,19 @@ func NewStore(dBConnection DBConnection) *Store {
 func (store *Store) GetCSVRows(filter *Filter, limit *int) (CSVRowReader, error) {
```
Comment (Matt-Guest) [inline observation/store.go:33]: Do mind expanding the godoc to include what happens with no dimensions?

**Comment by Matt-Guest** [comment]
Just minor comments

File: observation/store.go (hunk 2 above)
Comment (daiLlew) [reply observation/store.go:33]: 👍

**Review by Matt-Guest — APPROVED** [review]

**Pull request merged**